### PR TITLE
feat: qbd direct onboarding landing page

### PR DESCRIPTION
**Pull request by DhaaraniCIT**
### Description
feat: qbd direct onboarding landing page

## Clickup
https://app.clickup.com/t/86cwzcf6x
Screenshot 2024-11-04 at 5 38 38 PM
Screenshot 2024-11-04 at 5 38 48 PM

## Release Notes

- **New Features**
	- Enhanced dynamic branding in user interface content for improved personalization.
	- Added a method for updating workspace onboarding state, streamlining onboarding processes.
	- Introduced new components and properties in the onboarding landing page for better user experience.

- **Bug Fixes**
	- Improved logic for rendering UI elements related to integration connections.

- **Style**
	- Expanded Tailwind CSS configuration for more flexible styling options. 

- **Refactor**
	- Updated class names and properties to reflect branding changes and improve clarity.

### CODE REVIEW

File: src/app/branding/c1-contents-config.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ export const brandingConfig: BrandingConfiguration = config as BrandingConfigura
 export const c1Contents = {
     qbd_direct: {
         landing: {
-            contentText: 'Import chart of accounts and projects from NetSuite and export expenses from your Expense Management account.',
+            contentText: 'Import data from QuickBooks Desktop to Fyle and export expenses from Fyle to QuickBooks Desktop',
```
Comment (ashwin1111) [inline src/app/branding/c1-contents-config.ts:9]: don't use "Fyle" keyword in c1 file, use expense management / use brandName secret

File: src/app/core/models/db/workspaces.model.ts [hunk 2]
```diff
@@ -7,3 +7,7 @@ export type Workspace = {
   created_at: Date;
   updated_at: Date;
 }
+
+export type updateWorkspaceOnboardingStatePost = {
```
Comment (ashwin1111) [inline src/app/core/models/db/workspaces.model.ts:11]: WorkspaceOnboardingState

File: src/app/core/services/common/workspace.service.ts [hunk 3]
```diff
@@ -74,4 +73,8 @@ export class WorkspaceService {
   getWorkspaceGeneralSettings(): Observable<any> {
     return this.apiService.get(`/workspaces/${this.getWorkspaceId()}/settings/general/`, {});
   }
+
+  updateWorkspaceOnboardingState(payload: updateWorkspaceOnboardingStatePost): Observable<any> {
```
Comment (ashwin1111) [inline src/app/core/services/common/workspace.service.ts:77]: `Observable<Workspace>`

File: src/app/integrations/qbd-direct/qbd-direct-onboarding/qbd-direct-onboarding-landing/qbd-direct-onboarding-landing.component.ts [hunk 4]
```diff
@@ -1,12 +1,45 @@
...
 
+  appName: AppName = AppName.QBD_DIRECT;
+
+  brandingConfig = brandingConfig;
+
+  redirectLink = brandingKbArticles.topLevelArticles.QBD_DIRECT;
+
+  embedVideoLink = brandingDemoVideoLinks.onboarding.QBD_DIRECT;
+
+  isQbdDirectConnectionInProcess = false;
```
Comment (ashwin1111) [inline src/app/integrations/qbd-direct/qbd-direct-onboarding/qbd-direct-onboarding-landing/qbd-direct-onboarding-landing.component.ts:27]: `isQbdConnectionInProgress`

File: src/app/shared/components/helper/app-landing-page-header/app-landing-page-header.component.html [hunk 5]
```diff
@@ -40,9 +40,10 @@
                 <button *ngIf="isIntegrationConnected && (appName === AppName.BAMBOO_HR || appName === AppName.TRAVELPERK)" pButton type="button" class="p-button danger-outline" (click)="disconnect()">
                     {{ isConnectionInProgress ? 'Disconnecting' : 'Disconnect' }}
                 </button>
-                <button *ngIf="!isIntegrationConnected && (appName === AppName.QBD || appName === AppName.NETSUITE || appName === AppName.INTACCT || appName === AppName.SAGE300 || appName === AppName.BUSINESS_CENTRAL)" pButton type="button" class="p-button-raised" (click)="connect()">
+                <button *ngIf="!isIntegrationConnected && (appName === AppName.QBD || appName === AppName.NETSUITE || appName === AppName.INTACCT || appName === AppName.SAGE300 || appName === AppName.BUSINESS_CENTRAL || appName.includes(AppName.QBD_DIRECT))" pButton type="button" class="p-button-raised" (click)="connect()">
```
Comment (ashwin1111) [inline src/app/shared/components/helper/app-landing-page-header/app-landing-page-header.component.html:43]: why not `appName === AppName.QBD_DIRECT` ?

File: src/app/shared/components/helper/app-landing-page-header/app-landing-page-header.component.html [hunk 6]
```diff
@@ -40,9 +40,10 @@
...
                     {{buttonText}}
-                    <app-svg-icon *ngIf="appName === AppName.QBD && brandingFeatureConfig.isIconsInsideButtonAllowed" [svgSource]="'arrow-tail-right-medium'" [width]="'18px'" [height]="'18px'" [styleClasses]="'tw-pl-10-px tw-pt-2-px !tw-text-12-px'"></app-svg-icon>
+                    <app-svg-icon *ngIf="(appName === AppName.QBD || appName.includes(AppName.QBD_DIRECT)) && brandingFeatureConfig.isIconsInsideButtonAllowed" [svgSource]="'arrow-tail-right-medium'" [width]="'18px'" [height]="'18px'" [styleClasses]="'tw-pl-10-px tw-pt-2-px !tw-text-12-px'"></app-svg-icon>
```
Comment (ashwin1111) [inline src/app/shared/components/helper/app-landing-page-header/app-landing-page-header.component.html:45]: same as above

File: src/app/shared/components/helper/app-landing-page-header/app-landing-page-header.component.ts [hunk 7]
```diff
@@ -90,8 +90,7 @@ export class AppLandingPageHeaderComponent implements OnInit {
   }
 
   connect(): void {
-
-    if (this.appName === AppName.TRAVELPERK || this.appName === AppName.BUSINESS_CENTRAL || this.appName === AppName.BAMBOO_HR || this.appName === AppName.XERO) {
+    if (this.appName === AppName.TRAVELPERK || this.appName === AppName.BUSINESS_CENTRAL || this.appName === AppName.BAMBOO_HR || this.appName === AppName.XERO || this.appName.includes(AppName.QBD_DIRECT)) {
```
Comment (ashwin1111) [inline src/app/shared/components/helper/app-landing-page-header/app-landing-page-header.component.ts:93]: same as above

File: src/assets/logos/qbd-direct-logo.svg
Comment (ashwin1111) [inline src/assets/logos/qbd-direct-logo.svg:1]: use QBO logo from the repo present already

Comment (DhaaraniCIT) [reply src/assets/logos/qbd-direct-logo.svg:1]: QA wont affect, if we use different logo,

Comment (ashwin1111) [reply src/assets/logos/qbd-direct-logo.svg:1]: let's use this assets/logos/qbo.png

File: src/app/shared/components/helper/app-landing-page-header/app-landing-page-header.component.html (hunk 5 above)
Comment (DhaaraniCIT) [reply src/app/shared/components/helper/app-landing-page-header/app-landing-page-header.component.html:43]: this is due to the name, we use app name here as 'QuickBooks Desktop - Direct Integration'

File: src/app/shared/components/helper/app-landing-page-header/app-landing-page-header.component.html (hunk 6 above)
Comment (DhaaraniCIT) [reply src/app/shared/components/helper/app-landing-page-header/app-landing-page-header.component.html:45]: same as above

File: src/app/shared/components/helper/app-landing-page-header/app-landing-page-header.component.ts (hunk 7 above)
Comment (DhaaraniCIT) [reply src/app/shared/components/helper/app-landing-page-header/app-landing-page-header.component.ts:93]: same as above

File: src/app/core/services/common/workspace.service.ts (hunk 3 above)
Comment (DhaaraniCIT) [reply src/app/core/services/common/workspace.service.ts]: Workspace is a generic type, we will miss some properties, so using Workspace will create errors, while calling API we can use our model for that

Comment (DhaaraniCIT) [reply src/app/core/services/common/workspace.service.ts]: yes, but in Workspace model we are only having the common keys only, and we interface with workspace model and write the integration respective workspace mode. in the Workspace we don't have `onboarding_state` key. I am using it to store the onboarding state in local storage. that why

Comment (ashwin1111) [reply src/app/core/services/common/workspace.service.ts]: okay sorry, I meant QBDDirectWorkspace

Comment (DhaaraniCIT) [reply src/app/core/services/common/workspace.service.ts]: ok, cool

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**